### PR TITLE
Fix JS compilation to use the command 'compile js' instead of using snapshot names to invoke dart2js

### PR DESCRIPTION
**Pull request by a-siva**
Fix JS compilation to use the command 'compile js' instead of using snapshot names to invoke dart2js

The Dart SDK is switching all tools to use AOT snapshots instead of JIT snapshots and using snapshot names directly causes things to break.

### CODE REVIEW

File: packages/flutter_tools/lib/src/build_system/targets/web.dart [hunk 1]
```diff
@@ -170,7 +170,8 @@ class Dart2JSTarget extends Dart2WebTarget {
     final String platformBinariesPath = artifacts.getHostArtifact(HostArtifact.webPlatformKernelFolder).path;
     final List<String> sharedCommandOptions = <String>[
       artifacts.getArtifactPath(Artifact.engineDartBinary, platform: TargetPlatform.web_javascript),
-      artifacts.getArtifactPath(Artifact.dart2jsSnapshot, platform: TargetPlatform.web_javascript),
```
Comment (yjbanov) [inline packages/flutter_tools/lib/src/build_system/targets/web.dart:173]: Do we still need `Artifact.dart2jsSnapshot` for anything? If not, can you please also delete the enum value?

Comment (a-siva) [reply packages/flutter_tools/lib/src/build_system/targets/web.dart:173]: I have removed all references to the dart2js and dart2wasm snapshots.

Comment (a-siva) [reply packages/flutter_tools/lib/src/build_system/targets/web.dart:173]: PTAL

**Review by yjbanov — APPROVED** [review]
lgtm

**Pull request merged**